### PR TITLE
Fix GRANT ALL statement when executed on a non-global level

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Bug Fix

Changelog entry:
Fix GRANT ALL statement when executed on a non-global level.

### CODE REVIEW

**Comment by vitlibar** [comment]
@vzakaznikov Could you help fixing the testflows checks? `GRANT INTROSPECTION(x) ON ...` shows an error just as planned, so here the check is wrong. And please check https://github.com/ClickHouse/ClickHouse/issues/12600, it's not a bug, so XFails about it are not correct.

File: tests/testflows/rbac/tests/syntax/grant_privilege.py [hunk 1]
```diff
@@ -20,30 +20,30 @@ def setup(node):
...
-    ("TRUNCATE",("db0.table0","db0.*","*.*","tb0","*"), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_Truncate("1.0"))),
-    ("OPTIMIZE",("db0.table0","db0.*","*.*","tb0","*"), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_Optimize("1.0"))),
-    ("SHOW",("db0.table0","db0.*","*.*","tb0","*"), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_Show("1.0"))),
-    ("KILL QUERY",("*.*",), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_KillQuery("1.0"))),
-    ("ACCESS MANAGEMENT",("*.*",), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_AccessManagement("1.0"))),
-    ("SYSTEM",("db0.table0","db0.*","*.*","tb0","*"), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_System("1.0"))),
-    ("SOURCES",("*.*",), False, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_Sources("1.0"))),
-    ("ALL",("*.*",), True, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_All("1.0"))),
-    ("ALL PRIVILEGES",("*.*",), True, Requirements(RQ_SRS_006_RBAC_Grant_Privilege_All("1.0"))), #alias for all
+@Examples("privilege on allow_column allow_introspection", [
```
Comment (vitlibar) [inline tests/testflows/rbac/tests/syntax/grant_privilege.py:23]: @vzakaznikov I've changed this test a bit, could you check please that everything's ok?

**Comment by alexey-milovidov** [comment]
AST Fuzzer: #13981

File: tests/testflows/rbac/tests/syntax/grant_privilege.py (hunk 1 above)
Comment (alexey-milovidov) [reply tests/testflows/rbac/tests/syntax/grant_privilege.py:23]: Did not answer, let's assume it's ok.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**